examples/tests/valid/apps_with_params/nested_params/parameters.test.w

```diff
@@ -0,0 +1,23 @@
+struct Person {
+  name: str;
+  age: num;
+}
+
+struct House {
+  address: str;
+  residents: Array<Person>;
+}
+
+struct MyParams {
+  houses: Array<House>;
+}
+
+let app = nodeof(this).app;
+
+app.parameters.addSchema(MyParams.schema());
+
+let myParams = MyParams.fromJson(app.parameters.json());
```

Comment:
Any reason these two APIs can't be one?

```suggestion
let myParams = MyParams.fromJson(app.parameters.json(schema: MyParams.schema()));
```

Reply:
Im not sure I understand the implication of `paramters.json()` accepting a schema object. Does passing this schema in automatically add it to the applications parameters? It feels a bit off to have this accessor style method also be able to accept a parameter that mutates.

Reply:
What's the use case of decoupling these APIs? It seems like you would always call `addSchema()` and then immediately afterwards you'd call `json()` to obtain the values.

So we might as well just have a single API that allows you to read the parameters and validate the schema. Maybe it should be `parameters.read(schema: MyParams.schema())` or something like this.

Reply:
Okay so I went ahead and changed things to use `parameters.read()` which can be called with or without options. I think there is still a use case for just wanting to read all the parameters in without adding a schema to the registrar. 

so both work: `parameters.read()` returns all parameters back as Json, and `parameters.read(schema: Foo.schema())` will only return back the parameters that are part of that schema, but also ensure the schema is added to the registrar.